[Ready] Admin Tools 1: Modular Curses

josephpatrickgreen opened the pull request:
## About The Pull Request

Examples of curses:
-when a player is kissed shapeshift them into a frog or back to normal
-at nighttime make a player ugly, at dawn make them beautiful
-custom lycanthropy
-spontaneuos combustion upon orgasm
-shocked when cutting down a tree (fuck you dendor)
-inject beer into the bloodstream when attacked
-gib when beheaded

Changes the curse system  allowing admins to construct a custom curse for a player which persists semi-permanently.

1. Pick a trigger
	"on spawn",
	"on death",
	"on beheaded",
	"on dismembered",
	"on sleep",
	"on attack",
	"on receive damage",
	"on cast spell",
	"on spell or miracle target",
	//"on break wall/door/window",
	"on cut tree",
	//"on craft",
	"on kiss",
	"on kissed",
	"on orgasm",
	//"on bite",
	//"on jump",
	//"on climb",
	//"on swim",
	"on move",
	"on dawn",
	"on day",
	"on dusk",
	"on night"

2. Pick an effect:
	"buff or debuff",
	"remove trait",
	"add trait",
	"scream",
	"cry",
	"add reagent",
	"add arousal",
	"orgasm",
	"shrink sex organs",
	"enlarge sex organs",
	"add nausea",
	"clothesplosion",
	"slip",
	"arcyne prison",
	"make deadite",
	/*"make vampire",
	"make werewolf",*/
	"shock",
	"add fire stack",
	"cbt",
	//"easy ambush",
	//"difficult ambush",
	"explode",
	"shapeshift",
	"gib"

there are also options for cooldown, %chance, duration, reason etc etc.

Further, the curse can be flavored to divine, unholy, witchcraft, fey or mutation which changes the notifications
examples being:

**to self:**
A chill coils through your gut as you fall asleep. Your stomach twists.
You feel stained by something foul as you move. Your footing vanishes.
Old magic stirs beneath your skin as you invoke power. A new aspect awakens within you.
A shimmer brushes your senses as dawn breaks. Your blood feels altered.
A mocking giggle echoes in your skull as you are kissed. Heat eminates from your loins

**to others:**
A harsh blaze of light strikes NAME from above as their blow lands. Their stance subtly changes!
Shadows deepen around NAME as blood is drawn. Their face pales and their balance falters!
A brief glow outlines NAME's shape as wood splinters beneath their strike. Fabric bursts away from them!
A faint chorus of cruel laughter surrounds NAME with each step they take. They stumble and fall!
Soft snickering and rustling circle NAME as shadows deepen around them. A familiar quality disappears from them! 

## Testing Evidence
<img width="1027" height="820" alt="image" src="https://github.com/user-attachments/assets/a339ae1c-92ea-4f70-beca-a51e3831d513" />
<img width="305" height="282" alt="image" src="https://github.com/user-attachments/assets/d0aba18e-7c55-43e2-afab-b6ace949f0b5" />

<img width="381" height="330" alt="image" src="https://github.com/user-attachments/assets/ad46b22b-6f01-4964-8417-12d17250bf5b" />
<img width="339" height="276" alt="image" src="https://github.com/user-attachments/assets/79de7c31-c7ab-4e35-804a-8bd82a72251d" />
<img width="334" height="302" alt="image" src="https://github.com/user-attachments/assets/4c8f4a21-6748-482d-acfa-8e184d208f37" />
<img width="378" height="391" alt="image" src="https://github.com/user-attachments/assets/d9ed6867-0710-48b7-811a-d2863fba30fd" />
<img width="728" height="330" alt="image" src="https://github.com/user-attachments/assets/874b4ebd-a83c-4ca4-8120-8a568cbb003a" />


## Why It's Good For The Game
Adds another tool to give players a carrot or a stick.
a minor punishment to a player who should perhaps try something new
OR
boons to a player who did something cool in an event

a player could also ask you to make them the following curse. on time midnight give them the trait ugly. at noon, remove the trait ugly, just flavor shit.

another example is turning someone into a chicken at midnight and back at dawn

Elli-Skala commented:
I know this a admin only tool. But "on receiving mirical -> lesser mirical on swlf" doesn't cause super mega server death right?

josephpatrickgreen commented:
> I know this a admin only tool. But "on receiving mirical -> lesser mirical on swlf" doesn't cause super mega server death right?

I have removed the miracle specific effect and replaced with buff/debuff which could have the same effect

cynical24 commented:
this is so sovlfvl
